macOS make_move_iterator ambiguous call

When trying to compile the `Optimal_transport_demo` on macOS, the call to the function `compute_average_spacing` produces the error:

```bash
In file included from /Users/monet/Documents/fork/cgal/Optimal_transportation_reconstruction_2/test/Optimal_transportation_reconstruction_2/test_basic.cpp:1:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/vector:1959:30: error: call to
      'make_move_iterator' is ambiguous
    insert(__make_iter(__p), make_move_iterator(__v.begin()),
                             ^~~~~~~~~~~~~~~~~~
/Users/monet/Documents/fork/cgal/Spatial_searching/include/CGAL/Kd_tree.h:286:9: note: in instantiation of function template specialization
      'std::__1::vector<boost::iterators::transform_iterator<UF, std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default,
      boost::use_default>, std::__1::allocator<boost::iterators::transform_iterator<UF, std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>,
      boost::use_default, boost::use_default> > >::insert<CGAL::Prevent_deref<boost::iterators::transform_iterator<UF,
      std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default, boost::use_default> > >' requested here
    pts.insert(pts.end(), first, beyond);
        ^
/Users/monet/Documents/fork/cgal/Point_set_processing_3/include/CGAL/Point_set_processing_3/internal/Neighbor_query.h:105:7: note: in instantiation of
      function template specialization 'CGAL::Kd_tree<CGAL::Search_traits_adapter<boost::iterators::transform_iterator<UF,
      std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default, boost::use_default>,
      CGAL::Point_set_processing_3::internal::Neighbor_query<CGAL::Epick, const CGAL::Iterator_range<boost::iterators::transform_iterator<UF,
      std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default, boost::use_default> > &,
      CGAL::Identity_property_map<CGAL::Point_3<CGAL::Epick> > >::Deref_point_map, CGAL::Search_traits_3<CGAL::Epick> >,
      CGAL::Sliding_midpoint<CGAL::Search_traits_adapter<boost::iterators::transform_iterator<UF, std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>,
      boost::use_default, boost::use_default>, CGAL::Point_set_processing_3::internal::Neighbor_query<CGAL::Epick, const
      CGAL::Iterator_range<boost::iterators::transform_iterator<UF, std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default,
      boost::use_default> > &, CGAL::Identity_property_map<CGAL::Point_3<CGAL::Epick> > >::Deref_point_map, CGAL::Search_traits_3<CGAL::Epick> >,
      CGAL::Plane_separator<double> >, CGAL::Boolean_tag<true>, CGAL::Boolean_tag<true>
      >::Kd_tree<CGAL::Prevent_deref<boost::iterators::transform_iterator<UF, std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default,
      boost::use_default> > >' requested here
    , m_tree (iterator(m_points.begin()), iterator(m_points.end()), Splitter(), m_traits)
      ^
/Users/monet/Documents/fork/cgal/Point_set_processing_3/include/CGAL/compute_average_spacing.h:192:18: note: in instantiation of member function
      'CGAL::Point_set_processing_3::internal::Neighbor_query<CGAL::Epick, const CGAL::Iterator_range<boost::iterators::transform_iterator<UF,
      std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default, boost::use_default> > &,
      CGAL::Identity_property_map<CGAL::Point_3<CGAL::Epick> > >::Neighbor_query' requested here
  Neighbor_query neighbor_query (points, point_map);
                 ^
/Users/monet/Documents/fork/cgal/Point_set_processing_3/include/CGAL/compute_average_spacing.h:243:10: note: in instantiation of function template
      specialization 'CGAL::compute_average_spacing<CGAL::Sequential_tag, CGAL::Iterator_range<boost::iterators::transform_iterator<UF,
      std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default, boost::use_default> >, bool, CGAL::internal_np::all_default_t,
      CGAL::internal_np::No_property>' requested here
  return compute_average_spacing<ConcurrencyTag>
         ^
/Users/monet/Documents/fork/cgal/Optimal_transportation_reconstruction_2/test/Optimal_transportation_reconstruction_2/test_basic.cpp:18:9: note: in
      instantiation of function template specialization 'CGAL::compute_average_spacing<CGAL::Sequential_tag,
      CGAL::Iterator_range<boost::iterators::transform_iterator<UF, std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default,
      boost::use_default> > >' requested here
  CGAL::compute_average_spacing<CGAL::Sequential_tag>(
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iterator:1318:1: note: candidate function
      [with _Iter = boost::iterators::transform_iterator<UF, std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default,
      boost::use_default> *]
make_move_iterator(_Iter __i)
^
/usr/local/include/boost/move/iterator.hpp:173:42: note: candidate function [with It = boost::iterators::transform_iterator<UF,
      std::__1::__wrap_iter<CGAL::Point_3<CGAL::Epick> *>, boost::use_default, boost::use_default> *]
BOOST_MOVE_FORCEINLINE move_iterator<It> make_move_iterator(const It &it)
                                         ^
1 error generated.
make[2]: *** [CMakeFiles/test_basic.dir/test_basic.cpp.o] Error 1
make[1]: *** [CMakeFiles/test_basic.dir/all] Error 2
make: *** [all] Error 2
```

The problem comes from the `std::vector` insert method when trying to insert points from the range that is created on the fly that is the points are not stored anywhere but are rather produced upon arrival. The problem is an ambiguous call to the function `make_move_iterator` that can be chosen between `std::make_move_iterator` and `boost::make_move_iterator`.

## Source Code

I have created a minimal example that reproduces this issue:
https://gist.github.com/danston/fcc5b809cfd39652036b7b47f269f488

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): macOS
* Compiler: clang
* Release or debug mode: release and debug
* CGAL version: master
* Boost version: 1.75
* Other libraries versions if used (Eigen, TBB, etc.): Eigen, TBB.

--- a/Spatial_searching/include/CGAL/Kd_tree.h
+++ b/Spatial_searching/include/CGAL/Kd_tree.h
@@ -281,10 +281,8 @@ public:
   template <class InputIterator>
   Kd_tree(InputIterator first, InputIterator beyond,
           Splitter s = Splitter(),const SearchTraits traits=SearchTraits())
-    : traits_(traits),split(s), built_(false), removed_(false)
-  {
-    pts.insert(pts.end(), first, beyond);
-  }
+    : traits_(traits), split(s), pts(first, beyond), built_(false), removed_(false)
+  { }
 
   bool empty() const {
     return pts.empty();

